coriolis/utils.py

```diff
@@ -306,10 +307,13 @@ def exec_ssh_cmd(ssh, cmd, environment=None, get_pty=False):
         "environment %s: '%s'", remote_str, environment, cmd)
 
     _, stdout, stderr = ssh.exec_command(
-        cmd, environment=environment, get_pty=get_pty)
+        cmd, environment=environment, get_pty=get_pty, timeout=timeout)
+    try:
+        std_out = stdout.read()
+        std_err = stderr.read()
+    except socket.timeout:
+        raise exception.OSMorphingSSHOperationTimeout(cmd=cmd, timeout=timeout)
```

Comment:
`OSMorphingSSHOperationTimeout` is too specific an exception to raise here since this method can be also used to run non-OSMorphing commands.

Please make a separate `MinionMachineCommandTimeout` exception class and raise that here, and make the base `OSMorphingOperationTimeout` inherits from.
Afterwards, in the OSMorphing tools classes, use `try-except MinionMachineCommandTimeout: raise OSMorphing(SSH|WinRM)OperationTimeout(str(ex)) from ex`